crates/sui-node/src/admin.rs

```diff
@@ -37,22 +75,73 @@ pub fn start_admin_server(port: u16, filter_handle: FilterHandle) {
     });
 }
 
-async fn get_filter(Extension(filter_handle): Extension<FilterHandle>) -> (StatusCode, String) {
-    match filter_handle.get() {
+async fn get_filter(State(state): State<Arc<AppState>>) -> (StatusCode, String) {
+    match state.filter_handle.get() {
         Ok(filter) => (StatusCode::OK, filter),
         Err(err) => (StatusCode::INTERNAL_SERVER_ERROR, err.to_string()),
     }
 }
 
 async fn set_filter(
-    Extension(filter_handle): Extension<FilterHandle>,
+    State(state): State<Arc<AppState>>,
     new_filter: String,
 ) -> (StatusCode, String) {
-    match filter_handle.update(&new_filter) {
+    match state.filter_handle.update(&new_filter) {
         Ok(()) => {
             info!(filter =% new_filter, "Log filter updated");
             (StatusCode::OK, "".into())
         }
         Err(err) => (StatusCode::BAD_REQUEST, err.to_string()),
     }
 }
+
+async fn clear_override_protocol_upgrade_buffer_stake(
+    State(state): State<Arc<AppState>>,
+    _body: String,
+) -> (StatusCode, String) {
+    match state.node.clear_override_protocol_upgrade_buffer_stake() {
+        Ok(()) => (
+            StatusCode::OK,
+            "protocol upgrade buffer stake cleared".to_string(),
+        ),
+        Err(err) => (StatusCode::INTERNAL_SERVER_ERROR, err.to_string()),
+    }
+}
+
+async fn set_override_protocol_upgrade_buffer_stake(
+    State(state): State<Arc<AppState>>,
+    buffer_bps: String,
+) -> (StatusCode, String) {
+    let Ok(buffer_bps) = buffer_bps.parse::<u64>() else {
+        return (
+            StatusCode::BAD_REQUEST,
+            "argument must be a positive integer".to_string(),
+        );
+    };
+
+    match state
+        .node
+        .set_override_protocol_upgrade_buffer_stake(buffer_bps)
+    {
+        Ok(()) => (
+            StatusCode::OK,
+            format!("protocol upgrade buffer stake set to '{}'", buffer_bps),
+        ),
+        Err(err) => (StatusCode::INTERNAL_SERVER_ERROR, err.to_string()),
+    }
+}
+
+async fn force_close_epoch(
+    State(state): State<Arc<AppState>>,
+    _body: String,
```

Comment:
Not sure how hard is it to pass parameter such as epoch id here (and check if epoch store epoch matches epoch_id) before the call.
I am thinking that likely use case for this RPC is following - we want to close epoch early, so we ask validators to run `close_epoch` RPC.

Problem with this is that some validators might run it too late, when reconfiguration is done, and close the wrong epoch, if we don't have this check

Reply:
Agreed. Anything that could lead to a local fork if done at the wrong epoch should explicitly come with epoch di

Reply:
good point - this is an easy fix